temporalio/contrib/openai_agents/_temporal_model_stub.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from temporalio import workflow
+
+with workflow.unsafe.imports_passed_through():
+    from datetime import timedelta
+    from typing import Any, AsyncIterator, Sequence, cast
+
+    from agents import (
+        AgentOutputSchema,
+        AgentOutputSchemaBase,
+        ComputerTool,
+        FileSearchTool,
+        FunctionTool,
+        Handoff,
+        Model,
+        ModelResponse,
+        ModelSettings,
+        ModelTracing,
+        Tool,
+        TResponseInputItem,
+        WebSearchTool,
+    )
+    from agents.items import TResponseStreamEvent
+
+    from temporalio.contrib.openai_agents.invoke_model_activity import (
+        ActivityModelInput,
+        AgentOutputSchemaInput,
+        FunctionToolInput,
+        HandoffInput,
+        ModelTracingInput,
+        ToolInput,
+        invoke_model_activity,
+    )
+
+
+class _TemporalModelStub(Model):
+    """A stub that allows invoking models as Temporal activities."""
+
+    def __init__(self, model_name: str | None) -> None:
+        self.model_name = model_name
+
+    async def get_response(
+        self,
+        system_instructions: str | None,
+        input: str | list[TResponseInputItem],
+        model_settings: ModelSettings,
+        tools: list[Tool],
+        output_schema: AgentOutputSchemaBase | None,
+        handoffs: list[Handoff],
+        tracing: ModelTracing,
+        *,
+        previous_response_id: str | None,
+    ) -> ModelResponse:
+        def get_summary(input: str | list[TResponseInputItem]) -> str:
+            ### Activity summary shown in the UI
+            try:
+                max_size = 100
+                if isinstance(input, str):
+                    return input[:max_size]
+                elif isinstance(input, list):
+                    seq_input = cast(Sequence[Any], input)
+                    last_item = seq_input[-1]
+                    if isinstance(last_item, dict):
+                        return last_item.get("content", "")[:max_size]
+                    elif hasattr(last_item, "content"):
+                        return str(getattr(last_item, "content"))[:max_size]
+                    return str(last_item)[:max_size]
+                elif isinstance(input, dict):
+                    return input.get("content", "")[:max_size]
+            except Exception as e:
+                print(f"Error getting summary: {e}")
+            return ""
+
+        def make_tool_info(tool: Tool) -> ToolInput:
+            if isinstance(tool, FileSearchTool):
+                return cast(FileSearchTool, tool)
+            elif isinstance(tool, WebSearchTool):
+                return cast(WebSearchTool, tool)
```

Comment:
This is all a bit suspicious to me. `cast` doesn't actually do anything except signal to the type checker that a type is a certain type. I don't think this would survive over the serialization boundary.